client/js/controllers/MovieController.js

```diff
@@ -1,15 +1,43 @@
 var kitso = angular.module('kitso');
 
 kitso.controller('MovieController',
-['$scope', '$location', '$timeout', 'MovieService', 'WatchedService', '$routeParams', 'AuthService',
-function($scope, $location, $timeout, MovieService, WatchedService, $routeParams, AuthService) {
+['$scope', '$location', '$timeout', 'MovieService', 'WatchedService', 'RatedService', '$routeParams', 'AuthService',
+function($scope, $location, $timeout, MovieService, WatchedService, RatedService, $routeParams, AuthService) {
 
     MovieService.loadMovie($routeParams.movie_id)
         .then(() => {
           AuthService.getStatus().then(function(){
             $scope.user = AuthService.getUser();
             $scope.movie = MovieService.getMovie();
             $scope.release_date_formated = moment($scope.movie.release_date).format('YYYY');
+            $scope.updateRating(0);
+            RatedService.isRated($scope.user._id ,$routeParams.movie_id).then((rated) => {
+                $scope.movie.rated = rated;
+                if (! rated.rated_id){
+                  $scope.movie.rated = false;
+                  $scope.updateRating(0);
+                }else{
+                  RatedService.getAllRated($scope.user._id).then((all) => {
```

Comment:
O que é isso?

Reply:
Não vejo a necessidade dessa parte.
E quando for iterar por coisas, usa o forEach que é mais legivel.

Reply:
é pra quando vc abrir a página ele atualizar o rating, 0 ou a nota q vc tinha dado antes

Reply:
Mas pra que esse getAllRated se ele ja retorna o rated especifico?

Reply:
pq o rated q já tem é só um true/false e o id, eu preciso da nota